src/components/responsive-flyout/OverlayHeader.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+import * as React from 'react';
+
+import classNames from 'classnames';
+import CloseButton from './CloseButton';
+import './styles/OverlayHeader.scss';
+
+export type Props = {
+    /** Components to render in the header */
+    children?: React.Node,
+    /** Set className to the overlay header */
+    className?: string,
+    /** Set click handler when close button is clicked */
+    onCloseClick?: Function,
+};
+
+const OverlayHeader = (props: Props) => {
+    const { children, onCloseClick, className } = props;
+
+    const handleClick = event => {
+        event.preventDefault();
+        event.stopPropagation();
```

Comment:
This is cool. `stopPropagation` is new to me. Can help me understand, is this here to catch and essentially ignore click events? What does it do for us in the context of this fullscreen modal, as it were?

Reply:
I want to make sure that it does not conflict with Flyout/Overlay onclick handlers. It contains functionality that could break the OverlayHeader

Reply:
Is this actually needed? Since `OverlayHeader` is used within `Overlay`, `Flyout` should ignore any clicks just like it would for other content within the overlay.

Reply:
The problem is that Flyout does add a onClick handler to Overlay and there are behaviors associated with it. More specifically with closeOnClick

https://github.com/box/box-ui-elements/blob/a2df0ed670ec2205f777ed37833577046b073da5/src/components/flyout/Flyout.js#L405